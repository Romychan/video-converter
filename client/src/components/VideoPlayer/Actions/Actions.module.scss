.actions {
  margin-top: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.square {
  width: 35px;
  height: 35px;
  border-radius: 10px;
  background: var(--color-secondary);
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: 0.2s;

  &:hover {
    background: var(--color-secondary-hover);
  }
}

.playback {
  height: 12px;
  border: 0;
  border-color: transparent transparent transparent var(--color-primary);
  border-style: solid;
  border-width: 6px 0px 6px 10px;
  transition: 0.1s;

  &.paused {
    border-style: double;
    border-width: 0px 0px 0px 12px;
  }
}
