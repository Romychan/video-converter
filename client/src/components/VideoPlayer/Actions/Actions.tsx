import clsx from 'clsx';

import { toHHMMSS } from '@utils/toHHMMSS';

import styles from './Actions.module.scss';

interface IActionsProps {
  togglePlay: () => void;
  isPlaying: boolean;
  progress: number;
  duration: number;
}

export const Actions = ({
  togglePlay,
  isPlaying,
  progress,
  duration,
}: IActionsProps) => {
  return (
    <div className={styles.actions}>
      <button
        className={styles.square}
        onClick={togglePlay}
        data-testid="actions-button"
      >
        <div
          className={clsx(styles.playback, {
            [styles.paused]: isPlaying,
          })}
          data-testid="actions-playing"
        ></div>
      </button>

      <p className="text text_small">
        {toHHMMSS((progress / 100) * duration)} / {toHHMMSS(duration)}
      </p>
    </div>
  );
};
