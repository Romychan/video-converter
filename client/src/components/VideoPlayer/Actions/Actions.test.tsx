import { fireEvent, screen, render } from '@testing-library/react';
import { vi } from 'vitest';

import { Actions } from './Actions';

describe('Actions', () => {
  it('should render correctly', () => {
    const handleChange = vi.fn();

    render(
      <Actions
        isPlaying={true}
        progress={10}
        duration={50}
        togglePlay={handleChange}
      />,
    );

    const durationElement = screen.getByText(/00:00:50/i);
    const progressElement = screen.getByText(/00:00:05/i);
    const playingButtonElement = screen.getByTestId('actions-playing');

    expect(durationElement).toBeInTheDocument();
    expect(progressElement).toBeInTheDocument();
    expect(playingButtonElement.getAttribute('class')).toMatch(/paused/gi);
  });

  it('should togglePlay working', () => {
    const handleChange = vi.fn();

    render(
      <Actions
        isPlaying={true}
        progress={10}
        duration={50}
        togglePlay={handleChange}
      />,
    );
    const buttonElement = screen.getByTestId('actions-button');

    fireEvent.click(buttonElement);
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('should pause working', () => {
    const handleChange = vi.fn();

    render(
      <Actions
        isPlaying={false}
        progress={10}
        duration={50}
        togglePlay={handleChange}
      />,
    );

    const playingButtonElement = screen.getByTestId('actions-playing');

    expect(playingButtonElement.getAttribute('class')).not.toMatch(/paused/gi);
  });

  it('should match snapshot', () => {
    const handleChange = vi.fn();
    const container = render(
      <Actions
        isPlaying={true}
        progress={10}
        duration={50}
        togglePlay={handleChange}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
