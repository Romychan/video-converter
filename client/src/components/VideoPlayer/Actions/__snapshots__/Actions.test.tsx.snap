// Vitest Snapshot v1

exports[`Actions > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_actions_449f60"
      >
        <button
          class="_square_449f60"
          data-testid="actions-button"
        >
          <div
            class="_playback_449f60 _paused_449f60"
            data-testid="actions-playing"
          />
        </button>
        <p
          class="text text_small"
        >
          00:00:05
           / 
          00:00:50
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_actions_449f60"
    >
      <button
        class="_square_449f60"
        data-testid="actions-button"
      >
        <div
          class="_playback_449f60 _paused_449f60"
          data-testid="actions-playing"
        />
      </button>
      <p
        class="text text_small"
      >
        00:00:05
         / 
        00:00:50
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
